### PR TITLE
Fix/tokens nitpicks

### LINKED ISSUE #12905: EVM Tokens / Send form select
Send form does not respect Tokens / Hidden / Unregosnised

Screenshot 2024-06-16 at 19 24 30

### PR DESCRIPTION
**Pull request by AdamSchinzel**
## Description

Resolved some nitpicks from #12501 and one issue with tokens select (done some important refactor in `getTokens` so it was better to have in the same PR).

## Related Issue

Resolves #12905

### CODE REVIEW

File: packages/suite/src/components/suite/FormattedCryptoAmount.tsx [hunk 1]
```diff
@@ -90,7 +90,7 @@ export const FormattedCryptoAmount = ({
         <Container className={className}>
             {!!signValue && <Sign value={signValue} />}
             <Value data-test={dataTest}>{formattedValue}</Value>
-            {formattedSymbol && <BlurUrls text={' ' + formattedSymbol} />}
+            {formattedSymbol && <BlurUrls text={'\u00A0' + formattedSymbol} />}
```
Comment (tomasklim) [inline packages/suite/src/components/suite/FormattedCryptoAmount.tsx:93]: I like using `&nbsp;` but ok

File: packages/suite/src/views/wallet/send/components/Outputs/components/Amount/components/TokenSelect.tsx [hunk 2]
```diff
@@ -49,35 +47,31 @@ export const buildTokenOptions = (
...
-    if (tokens) {
-        const unknownTokens: Option['options'] = [];
-        const hasCoinDefinitions = getNetworkFeatures(symbol).includes('coin-definitions');
-
-        tokens.forEach(token => {
-            if (new BigNumber(token?.balance || '').eq('0')) {
-                return;
-            }
+    if (accountTokens) {
+        const tokens = getTokens(accountTokens, symbol, isDebug, coinDefinitions);
```
Comment (tomasklim) [inline packages/suite/src/views/wallet/send/components/Outputs/components/Amount/components/TokenSelect.tsx]: does it make sense to have `isDebug` here? i dont think so

**Review by tomasklim — APPROVED** [review]
LGTM, just I would remove `isDebug` from token select if you agree

File: packages/suite/src/views/wallet/send/components/Outputs/components/Amount/components/TokenSelect.tsx (hunk 2 above)
Comment (AdamSchinzel) [reply packages/suite/src/views/wallet/send/components/Outputs/components/Amount/components/TokenSelect.tsx]: Removed it and set to `false` by default

**Pull request merged**